### ISSUE TITLE
mockDestination create wrong Destination for OAuth2ClientCredentials authentification type

### ISSUE THREAD

**Issue by nazarkulyk** [SAP/cloud-sdk-js#1207]
Currently mockDestination can create only 2 Types of Authentications: 'No' and ''.

Please extend method to support other AuthenticationTypes.

Example:

systems.json

```json
{
    "systems": [
        {
            "alias": "test-api",
            "uri": "https://myapidestination.local/v1.0/",
            "authentication": "OAuth2ClientCredentials"
        }
    ]
}
```

credentials.json

```json
{
    "credentials": [
        {
            "alias": "test-api",
            "clientId": "xxxx-xxx-xxxxx",
            "clientSecret": "xxxxxxxxxxxxxx"
        }
    ]
}
```

code to get Destination:

```js
const { mockTestDestination } = require('@sap-cloud-sdk/test-util');
const { getDestination } = require('@sap-cloud-sdk/core');

console.log(getDestination('test-api'));
```

**Comment by jjtang1985** [comment]
Hi @artemkovalyov ,
for me this is a feature request and it's about a convenient tool when developing unit tests.

**Comment by nazarkulyk** [comment]
@jjtang1985 
Basic Auth and ClientCredentials for oAuth2 is pretty same secure level. Don't understand why for mockDestination only limited set of destinations should be supported.
If its feature request its fine for me too.

**Comment by marikaner** [comment]
Hey @nazarkulyk, we will discuss this internally and will come back to you once we have a clearer picture, whether and when we can tackle this. Do you have a workaround that you can work with for now?

**Comment by nazarkulyk** [comment]
@marikaner yes, i do fake some coding to get stuff work.

**Comment by marikaner** [comment]
Hey @nazarkulyk, I just looked at this issue after a long time. We are considering to improve test utils in the future, but this is not on our current roadmap. 
We discussed this issue extensively in the past and decided that mocking entire authentication mechanisms (beyond the obvious simple ones) is not worth the effort if there is no high user demand.
For most testing cases against mocked systems NoAuth and Basic is probably enough, as those tests should cover business logic not authentication. As you even have a workaround, I am going to close this issue. Thank you for bringing it up, though!
